.github/workflows/scripts/e2e-maven-push.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# shellcheck source=/dev/null
+source "./.github/workflows/scripts/e2e-utils.sh"
+
+# This script bumps the maven package's version number, commits it, and pushes to
+# the repository.
+
+branch=$(e2e_this_branch)
+
+# NOTE: We can't simply push from $branch because it is occaisonally reset to
+# the main branch. We need to maintain the version number in pom.xml
+# because you cannot overwrite a version in maven. Instead we commit to main,
+# set the tag, reset $branch and push both main and $branch.
+echo "GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}"
+gh repo clone "${GITHUB_REPOSITORY}" -- -b maven-e2e-temp
+repo_name=$(echo "$GITHUB_REPOSITORY" | cut -d '/' -f2)
+cd ./"$repo_name"
+
+git config --global user.name github-actions
+git config --global user.email github-actions@github.com
+
+# Set the remote url to authenticate using the token.
+git remote set-url origin "https://github-actions:${GH_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
+
+package_dir="${PACKAGE_DIR}" # specified in the e2e test yaml
+
+cd "${package_dir}"
+
+# Get the new version
+artifact_tag=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+
+# version_major prints the major version number.
+# Expects a string like '1.19.7'
+# version_major returns "1" if the input is '1.19.7'
+version_major() {
```

Comment:
Can you avoid redefining these functions and use the existing one instead from https://github.com/slsa-framework/example-package/blob/main/.github/workflows/scripts/e2e-utils.sh#L121?

if mvn returns versions of type `x.y.z`, can we prepend `v` to artifact_tag to be able to re-use existing functions?

Reply:
We can prepend `v` to the version, but it is not a practice I have seen "in the wild". Let me know if we should change nonetheless

Reply:
Maven Central may have restrictions with this, but we are not pushing to Maven Central in this test.

Reply:
i just meant to prepend `v` to be able to re-use the existing functions from https://github.com/slsa-framework/example-package/blob/main/.github/workflows/scripts/e2e-utils.sh#L121, which expect `v` I think. If it does not expect a `v`, then we should be able to re-use them.

Reply:
Done.